indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/WorkerSelectUtils.java

```diff
@@ -95,6 +95,79 @@ public static ImmutableWorkerInfo selectWorker(
     }
   }
 
+  /**
+   * Helper for {@link WorkerSelectStrategy} implementations.
+   *
+   * @param allWorkers     map of all workers, in the style provided to {@link WorkerSelectStrategy}
+   * @param workerTierSpec worker tier spec, or null
+   * @param workerSelector function that receives a list of eligible workers: version is high enough, worker can run
+   *                       the task, and worker satisfies the worker tier spec. may return null.
+   *
+   * @return selected worker from "allWorkers", or null.
+   */
+  @Nullable
+  public static ImmutableWorkerInfo selectWorker(
+      final Task task,
+      final Map<String, ImmutableWorkerInfo> allWorkers,
+      final WorkerTaskRunnerConfig workerTaskRunnerConfig,
+      @Nullable final WorkerTierSpec workerTierSpec,
+      final Function<ImmutableMap<String, ImmutableWorkerInfo>, ImmutableWorkerInfo> workerSelector
+  )
+  {
+    // Workers that could potentially run this task, ignoring workerTierSpec.
+    final Map<String, ImmutableWorkerInfo> runnableWorkers = allWorkers
+        .values()
+        .stream()
+        .filter(worker -> worker.canRunTask(task)
+                          && worker.isValidVersion(workerTaskRunnerConfig.getMinWorkerVersion()))
+        .collect(Collectors.toMap(w -> w.getWorker().getHost(), Function.identity()));
+
+    // select worker according to worker tier spec
+    if (workerTierSpec != null && workerTierSpec.getTierMap() != null) {
+      final WorkerTierSpec.TierConfig tierConfig = workerTierSpec.getTierMap()
+                                                                 .getOrDefault(
+                                                                     task.getType(),
+                                                                     new WorkerTierSpec.TierConfig(null, null)
```

Comment:
why create a TierConfig with nulls when we could simply next check if(tierConfig == null) and do the default action which would happen anyway.

Reply:
Fixed.